File: ohara-kafka/src/test/scala/com/island/ohara/kafka/connector/SimpleRowSinkTask.scala
package com.island.ohara.kafka.connector

import com.island.ohara.common.data.{Row, Serializer}
import com.island.ohara.common.util.CloseOnce
import com.island.ohara.kafka.Producer
import com.island.ohara.kafka.connector.Constants._

/**
  * Used for testing.
  */
class SimpleRowSinkTask extends RowSinkTask {
  private[this] var config: TaskConfig = _
  private[this] var outputTopic: String = _
  private[this] var producer: Producer[Array[Byte], Row] = _
  override def _start(props: TaskConfig): Unit = {
    this.config = props
    outputTopic = config.options(OUTPUT)
    producer = Producer.builder().brokers(config.options(BROKER)).build(Serializer.BYTES, Serializer.ROW)
  }

  override def _put(records: Seq[RowSinkRecord]): Unit = {
    records.foreach(r => {
      producer.sender().key(r.key).value(r.row).send(outputTopic)
    })
  }

  override def _stop(): Unit = CloseOnce.close(producer)
}

File: ohara-kafka/src/test/java/com/island/ohara/kafka/connector/SimpleRowSinkTask.java
/*
 * Copyright 2019 is-land
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.island.ohara.kafka.connector;

import static com.island.ohara.kafka.connector.Constants.BROKER;
import static com.island.ohara.kafka.connector.Constants.OUTPUT;

import com.island.ohara.common.data.Row;
import com.island.ohara.common.data.Serializer;
import com.island.ohara.kafka.Producer;
import java.util.List;

/** Used for testing. */
public class SimpleRowSinkTask extends RowSinkTask {
  private TaskConfig config = null;
  private String outputTopic = null;
  private Producer<byte[], Row> producer = null;

  @Override
  protected void _start(TaskConfig props) {
    this.config = props;
    outputTopic = config.options().get(OUTPUT);
    producer =
        Producer.builder()
            .connectionProps(config.options().get(BROKER))
            .build(Serializer.BYTES, Serializer.ROW);
  }

  @Override
  protected void _stop() {
    try {
      producer.close();
    } catch (Exception e) {
      e.printStackTrace();
    }
  }

  @Override
  protected void _put(List<RowSinkRecord> records) {
    records.forEach(r -> producer.sender().key(r.key()).value(r.row()).send(outputTopic));
  }
}
